src/main/kotlin/no/nav/tilleggsstonader/sak/statistikk/behandling/BehandlingsstatistikkService.kt

```diff
@@ -0,0 +1,205 @@
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.dto.TotrinnkontrollStatus
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDateTime
+import java.util.UUID
+
+@Service
+class BehandlingsstatistikkService(
+    private val behandlingsstatistikkProducer: BehandlingKafkaProducer,
+    private val behandlingService: BehandlingService,
+    private val oppgaveService: OppgaveService,
+    private val personService: PersonService,
+    private val totrinnskontrollService: TotrinnskontrollService,
+) {
+
+    @Transactional
+    fun sendBehandlingstatistikk(
+        behandlingId: UUID,
+        hendelse: Hendelse,
+        hendelseTidspunkt: LocalDateTime,
+        gjeldendeSaksbehandler: String?,
+        oppgaveId: Long?,
+        behandlingMetode: BehandlingMetode?,
+    ) {
+        val behandlingDVH = mapTilBehandlingDVH(
+            behandlingId,
+            hendelse,
+            hendelseTidspunkt,
+            gjeldendeSaksbehandler,
+            oppgaveId,
+            behandlingMetode,
+        )
+        behandlingsstatistikkProducer.sendBehandling(behandlingDVH)
+    }
+
+    private fun mapTilBehandlingDVH(
+        behandlingId: UUID,
+        hendelse: Hendelse,
+        hendelseTidspunkt: LocalDateTime,
+        gjeldendeSaksbehandler: String?,
+        oppgaveId: Long?,
+        behandlingMetode: BehandlingMetode?,
+    ): BehandlingDVH {
+        val saksbehandling = behandlingService.hentSaksbehandling(behandlingId)
+        val sisteOppgaveForBehandling = finnSisteOppgaveForBehandlingen(behandlingId, oppgaveId)
+        val henvendelseTidspunkt = finnHenvendelsestidspunkt(saksbehandling)
+        val strengtFortroligAdresse = evaluerAdresseBeskyttelseStrengtFortrolig(saksbehandling.ident)
+        val saksbehandlerId = finnSaksbehandler(hendelse, totrinnskontrollService, gjeldendeSaksbehandler, behandlingId)
+        val beslutterId = utledBeslutterId(hendelse, behandlingId)
+        val relatertEksternBehandlingId: String? =
+            saksbehandling.forrigeBehandlingId?.let { behandlingService.hentSaksbehandling(it).eksternId.toString() }
+
+        return BehandlingDVH(
+            behandlingId = saksbehandling.eksternId.toString(),
+            behandlingUuid = behandlingId.toString(),
+            sakId = saksbehandling.eksternId.toString(),
+            aktorId = saksbehandling.ident,
+            registrertTid = saksbehandling.opprettetTid.atZone(europeOslo)
+                ?: henvendelseTidspunkt.atZone(europeOslo),
+            endretTid = hendelseTidspunkt.atZone(europeOslo),
+            tekniskTid = zonedNow(),
+            behandlingStatus = hendelse.name,
+            opprettetAv = maskerVerdiHvisStrengtFortrolig(
+                strengtFortroligAdresse,
+                saksbehandlerId,
+            ),
+            saksnummer = saksbehandling.eksternFagsakId.toString(),
```

Comment:
Jeg skjønner ikke helt forskjellen på `sakId` og `saksnummer`. Er det samme? Når jeg saksnummer/id tenker jeg i hvert fall fagsak 🤓

Reply:
sakId: "Hvis fagsystemet har et forhold til sak, skal saks-IDen oppgis her."
saksnummer: "Saksnummer som brukes av saksbehandler for oppslag i vedtaksløsningen."

Hva blir riktig her tenker du? 🤷‍♂️

Reply:
Såvidt jeg kan se har vi ikke en "lettlest" id, bare masse UUID-er